src/SimpleJson/Serialization.php

```diff
@@ -170,3 +170,96 @@ function convertToSimpleJson($obj) {
         return $obj;
 } 
 
+function convertType($json, $cls)
+{
+    $class_vars = get_class_vars($cls);
+    if (count($class_vars) == 0) {
+        return new $cls();
+    } elseif (count($class_vars) == 1) {
+        $fieldName =key($class_vars);
+        $method = "get_{$fieldName}_Type";
+        $field = convertFromSimpleJson($json, $cls::$method());
+        return new $cls($field);
+    } else {
+        $fields = [];
+        foreach ($class_vars as $field => $_) {
+            $method = "get_{$field}_Type";
+            $fields[$field] = convertFromSimpleJson($json->$field, $cls::$method());
+        }
+        return new $cls(...$fields);
+    }
+}
+
+function convertFromSimpleJson($json, $cls)
+{
+    if (is_null($json)) {
+        return null;
+    }
+
+    if ($cls == 'String') {
+        return $json;
+    }
+    if ($cls == 'Int32') {
+        return $json;
+    }
+
+    if (is_array($cls)) {
+        if ($cls[0] == "List") {
+            $result = [];
+            foreach ($json as $item) {
+                $result[] = convertFromSimpleJson($item, $cls[1]);
+            }
+            return FSharpList::ofArray($result);
+        } elseif ($cls[0] == "Set") {
+            $result = [];
+            $itemCls = $cls[1];
+            foreach ($json as $item) {
+                $result[] = convertFromSimpleJson($item, $itemCls);
+            }
+            return Set::ofSeq($result, ['Compare' => 'Util::compare']);
```

Comment:
@thinkbeforecoding Is this right? If I'm ready the code right, it should infer the correct compare from using this.

Reply:
Seem ok!